File: src-tauri/src/shared_types/command.rs

// use serde::{Deserialize, Serialize};
// #[derive(Debug, Serialize, Deserialize, Clone)]
// pub struct CommandEnvelope {
//     pub kind: String,
//     pub id: String,

//     #[serde(flatten)]
//     pub command: CommandBody,
// }

// #[derive(Debug, Serialize, Deserialize, Clone)]
// #[serde(tag = "moduletype", content = "payload", rename_all = "snake_case")]
// pub enum CommandBody {
//     Led(LedCommand),
//     ClusterLeds(ClusterCommand),
//     // Spelled to match the firmware's `ModuleCommand::Servo` wire tag.
//     Servo(ServoCommand),
// }


// #[derive(Debug, Serialize, Deserialize, Clone)]
// #[serde(tag = "command", rename_all = "snake_case")]
// pub enum LedCommand {
//     SetState { state: u32 },
//     Toggle,
// }

// #[derive(Debug, Serialize, Deserialize, Clone)]
// #[serde(tag = "command", rename_all = "snake_case")]
// pub enum ServoCommand {
//     SetAngle { angle: i32 },
//     SetMinPivot { min_pivot: i32 },
//     SetMaxPivot { max_pivot: i32 },
// }

// #[derive(Debug, Serialize, Deserialize, Clone)]
// #[serde(tag = "command", rename_all = "snake_case")]
// pub enum ClusterCommand {
//     ToggleAll,
//     SetAll { state: u32 },
//     Toggle { id: String, state: u32 },
//     SetState { id: String, state: u32 },
// }

File: src/Hook/Command.ts
// import z from "zod";




// export const cmdBase = {
//   kind: z.literal("CMD"),
//   id: z.string(),
// };


// export const LedCommandTypeScheme = z.discriminatedUnion("command", [
//   z.object({
//     command: z.literal("set_state"),
//     state: z.number(),
//   }),

//   z.object({
//     command: z.literal("toggle"),
//   }),
// ]);

// export const ServoCommandTypeScheme = z.discriminatedUnion("command", [
//   z.object({
//     command: z.literal("set_angle"),
//     angle: z.number(),
//   }),

//   z.object({
//     command: z.literal("set_min_pivot"),
//     min_pivot: z.number(),
//   }),

//   z.object({
//     command: z.literal("set_max_pivot"),
//     max_pivot: z.number(),
//   }),
// ]);

// export const ClusterCommandTypeScheme = z.discriminatedUnion("command", [
//   z.object({
//     command: z.literal("toggle_all"),
//   }),

//   z.object({
//     command: z.literal("set_all"),
//     state: z.number(),
//   }),

//   z.object({
//     command: z.literal("toggle"),
//     id: z.string(),
//     state: z.number(),
//   }),

//   z.object({
//     command: z.literal("set_state"),
//     id: z.string(),
//     state: z.number(),
//   }),
// ]);

// export const   SerialCMDLed =z.object({
//     ...cmdBase,
//     moduletype: z.literal("led"),
//     payload: LedCommandTypeScheme,
//   })

// export const SerialCMDServo = z.object({
//   ...cmdBase,
//   moduletype: z.literal("servo"),
//   payload: ServoCommandTypeScheme,
// })

// export const SerialCMDClusterLeds = z.object({
//   ...cmdBase,
//   moduletype: z.literal("cluster_leds"),
//   payload: ClusterCommandTypeScheme,
// })

// export const SerialCMDScheme = z.discriminatedUnion("moduletype", [
//   SerialCMDLed,
//   SerialCMDServo,
//   SerialCMDClusterLeds,
// ]);

// export type SerialCMDType = z.infer<typeof SerialCMDScheme>;
